static/app/views/alerts/rules/issue/ticketRuleModal.spec.tsx

```diff
@@ -149,28 +150,73 @@ describe('ProjectAlerts -> TicketRuleModal', function () {
       await submitSuccess();
     });
 
-    it('should ignore error checking when default is empty array', function () {
-      renderComponent({
-        otherField: {
-          label: 'Labels',
-          required: false,
-          choices: [['bug', `bug`]],
-          default: [],
-          type: 'select',
-          multiple: true,
-          name: 'labels',
-        },
+    it('should ignore error checking when default is empty array', async function () {
+      renderComponent(undefined, {
+        label: 'Labels',
+        required: false,
+        choices: [['bug', `bug`]],
+        default: undefined,
+        type: 'select',
+        multiple: true,
+        name: 'labels',
       });
       expect(
         screen.queryAllByText(`Could not fetch saved option for Labels. Please reselect.`)
       ).toHaveLength(0);
+
+      await selectEvent.select(screen.getByRole('textbox', {name: 'Issue Type'}), 'Epic');
+      await selectEvent.select(screen.getByRole('textbox', {name: 'Labels'}), 'bug');
+      await submitSuccess();
     });
 
-    it('should persist values when the modal is reopened', async function () {
+    it('should persist single select values when the modal is reopened', async function () {
       renderComponent({data: {reporter: 'a'}});
       await submitSuccess();
     });
 
+    it('should persist multi select values when the modal is reopened', async function () {
+      renderComponent(
+        {data: {components: ['a', 'c']}},
+        {
+          name: 'components',
+          label: 'Components',
+          default: undefined,
+          type: 'select',
+          multiple: true,
+          required: true,
+          choices: [
+            ['a', 'a'],
+            ['b', 'b'],
+            ['c', 'c'],
+          ],
+        }
+      );
+      await submitSuccess();
+    });
+
+    it('should not persist value when unavailable in new choices', async function () {
+      renderComponent();
+      await selectEvent.select(screen.getByRole('textbox', {name: 'Reporter'}), 'a');
+
+      addMockConfigsAPICall({
+        label: 'Reporter',
+        required: true,
+        choices: [['b', 'b']],
```

Comment:
'b' here twice

Reply:
That's how choices is supposed to look unfortunately 😅

Reply:
Oh oof lol ok